specs/Alert_Notifications.adoc

```diff
@@ -0,0 +1,189 @@
+
+The present alerting system monitors all the events and retrieves alert specific events.
+The alert specific events are processed and stored again in etcd under /alerting/alerts/
+as alerts recognized by tendrl. This leads to excess read and writes to etcd.
+Present implementation causes the alerting component to process events even
+when it can be done by a separate component before inserting the event in etcd.
+
+== Use Cases
+
+Goal: Notify user in case of any alert event.
+
+Actors:
+
+1) User
+
+Steps:
+
+* Node Agent stores alerting events in etcd.
+
+* These events will be fetched by the Notifier.
+
+* The alerting event is to be sent to respective user via configured
+  notification channels.
+
+== Proposed change
+
+
+Present Alert system(https://github.com/Tendrl/alerting) will be renamed as “Notifier”
+
+The main functionality of Notifier would be to:
+
+Monitor etcd only for alert specific events.
+
+Sending alert notifications to configurable receiving end-points.
+
+Presently the alerting module processes all events under messages\events
+in etcd and looks for events having priority as “notice” which signifies they are
+alerts. Instead of processing all events an AlertManager will be configured to
+receive just alerting events and notify users about these events.
+
+For this the notifier must receive only the alerting events.
+This can be achieved in 2 ways:
+
+* All alerting events will be stored in etcd under alerting/alerts/{cluster,node} by
+  the node-agent as per the scope of the event (cluster level scope or node level scope).
+  A watcher(AlertWatcher) will be configured to poll etcd (alerting/alerts) for any new
+  events. On detecting new events the evetns will be passed to notifier which would
+  retrieve payload from the event, parse it and using the configured notification channel
+  would notify the users of these events.
+
+ Advantage:
+ 
+     ** Simple architecture and easy to implement.
+
+  Drawback:
+ 
+      ** Polling etcd for new alerts is not efficient.
+ 
+* To avoid polling etcd, node agent instead of just logging the events to
+  alerting/alerts/{cluster,node} could also pass events to Notifier directly.
+  Notifier can extract the payload,parse it and using the configured notification channel
+  would notify the users of these events.   
+
+    Advantage:
+ 
+     ** Avoids polling etcd for new alerts.
+
+
+The Notifier will notify the user of the events via configured notification channels
+like SMTP.
+
+=== Alternatives:
+
+None
+
+
+=== Data model impact:
+
+ No changes in existing structure.
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+None
+
+==== Notifications/Monitoring impact:
+
+None
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+Node-agent must be modified to insert the events received by Monitoring
+Integration in etcd under alerting/alerts/ instead of messages/events/.
+
+Monitoring integration adds scope ( global, cluster specific or node specific)
```

Comment:
Monitoring integration is just one of the components that generates an alert condition from an event. Such would also be the case for node agent, {ceph,gluster}-integration. All of these systems are responsible for receiving events from their underlying systems.